### PR TITLE
Add image support for adoptopenjdk

### LINKED ISSUE #136: Move to AdoptOpenJDK?
Would it make sense to move to AdoptOpenJDK due to Oracles lack of updates and their licensing? AdoptOpenJDK is still updated and also free.

### PR DESCRIPTION
**Pull request by cwholmes**
Fixes #136 

This does not move the current images to adoptopenjdk, but it instead adds images to be built and deployed with adoptopenjdk as the base

### CODE REVIEW

File: adoptopenjdk-11/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ENV BASE_URL=https://archive.apache.org/dist/maven/maven-3/${MAVEN_VERSION}/binaries
+
+RUN mkdir -p /usr/share/maven/; \
+    curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz; \
+    echo "${SHA}  /tmp/apache-maven.tar.gz" | sha512sum -c -; \
+    tar -xzf /tmp/apache-maven.tar.gz -C /usr/share/maven --strip-components=1
+
+COPY settings-docker.xml /usr/share/maven/ref/
+
+FROM adoptopenjdk/openjdk11:x86_64-alpine-jdk-11.0.5_10-slim
```
Comment (carlossg) [inline adoptopenjdk-11/Dockerfile]: better to use the less specific image so it doesn't need to be updated continuously

**Comment by carlossg** [comment]
Actually, there is an official `adoptopenjdk` docker image that can be used instead https://hub.docker.com/_/adoptopenjdk

**Comment by cwholmes** [comment]
It seems the official images are only ubuntu/windows based and the unofficial images support alpine. Would it be reasonable to have both? adoptopenjdk-8 and adoptopenjdk-8-alpine/adoptopenjdk-11 and adoptopenjdk-11-alpine one that could be hosted as an official image and the other that would not?

Thanks for looking at this. I will make the other suggestions and update the PR.

**Comment by cwholmes** [comment]
I have made the updates to use the official adoptopenjdk ubuntu images. I will wait for your suggestions on whether or not to add the alpine versions. The difference here is going to be an image base size of 90MB for alpine and 305MB for the jdk 8 versions. The sizes would be slightly larger for the jdk11 types.

**Comment by carlossg** [comment]
`mvn-entrypoint.sh` is missing the executable bit, everything else looks good

**Comment by cwholmes** [comment]
I'm not sure what you mean by the executable bit? Could you please clarify?

**Comment by carlossg** [comment]
`chmod +x adoptopenjdk-*/mvn-entrypoint.sh`

you didn't gave permissions to push to this PR otherwise I could just fix it myself

you can run the tests

```
TAG=adoptopenjdk-8 bats tests 
TAG=adoptopenjdk-11 bats tests
```

**Comment by cwholmes** [comment]
Updated with executable.

**Comment by carlossg** [comment]
thanks!

**Pull request merged**